src/sentry/migrations/0021_hide_environment_none_20191126.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+def hide_environment_none(apps, schema_editor):
+    """
+    Hide environments that are named none, since they're blacklisted and no longer can be created.
+
+    We should iterate over each environment row individually in python instead so that we don't lock the DB up. This is
+    far slower but much safer
+    """
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+    for env in RangeQuerySetWrapperWithProgressBar(Environment.objects.all()):
```

Comment:
Given that you have a condition on name shouldn't that be moved into the where clause? That would make the migration idempotent and use fewer resources.

Reply:
I suggested not to do this since we don't have an index. It'd shift the resources onto the database, which is not where I want excess resources.

But since we're going deep here, it seems that Postgres is doing what I wouldn't expect it to be doing. It's using a different index that isn't related, and that index seems fine.

```
getsentry=# explain analyze select * from sentry_environment where name='none';
                                                                                  QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Index Scan using sentry_environment_organization_id_6c9098a3d53d6a9a on sentry_environment  (cost=xxx rows=xxx width=xxx) (actual time=xxx rows=xxx loops=1)
   Index Cond: ((name)::text = 'none'::text)
 Planning time: 0.147 ms
 Execution time: 15.652 ms
(4 rows)
```

So I guess we could. This wouldn't be bad at all. I was just assuming and being safe. Ideally we'd all learn how to make these kinds of assessments.

Reply:
:+1: Thanks for the help, I'll update the migration docs to include doing a similar check in the future.